src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -176,43 +236,49 @@ public function indexAction(Request $request)
      */
     public function createAction(Request $request)
     {
-        $this->isGrantedOr403('create');
+        $configuration = $this->configurationFactory->create($this->metadata, $request);
 
-        $resource = $this->createNew();
-        $form = $this->getForm($resource);
+        $this->isGrantedOr403($configuration, ResourceActions::CREATE);
 
-        if ($request->isMethod('POST') && $form->submit($request)->isValid()) {
-            $resource = $this->domainManager->create($form->getData());
+        $resource = $this->createNew($configuration);
+        $form = $this->createForm($configuration, $resource);
 
-            if ($this->config->isApiRequest()) {
-                if ($resource instanceof ResourceEvent) {
-                    throw new HttpException($resource->getErrorCode(), $resource->getMessage());
-                }
+        if ($request->isMethod('POST') && $form->submit($request)->isValid()) {
+            $this->eventDispatcher->dispatch(GenericResourceEvents::PRE_CREATE, new GenericResourceEvent($resource, $this->metadata, $configuration, ResourceActions::CREATE));
+            $event = $this->eventDispatcher->dispatchResourceEvent(ResourceEvents::PRE_CREATE, $resource);
```

Comment:
Generic event is fired for all resources. You can listen to ``sylius.resource.pre_create`` and you will get it for every resource. This is how I handle flashes for example. Resource specific events are... well, resource specific. :)